GeneralsMD/Code/GameEngine/Include/Common/FileSystem.h

```diff
@@ -131,7 +131,7 @@ class FileSystem : public SubsystemInterface
 	void update();
 
 	File* openFile( const Char *filename, Int access = 0 );		///< opens a File interface to the specified file
-	Bool doesFileExist(const Char *filename) const;								///< returns TRUE if the file exists.  filename should have no directory.
+	Bool doesFileExist(const Char *filename, Bool cacheAbsence = TRUE) const;	///< returns TRUE if the file exists.  filename should have no directory.
```

Comment:
Perhaps you can review also all other code that calls `doesFileExist` and check if anything else writes files after `doesFileExist` test and then fix that as well (in separate changes).

Basically after a new file is written out and there is a code path that calls `FileSystem::doesFileExist` for it, then we better tell the FileSystem that we have written it. Likewise, we perhaps should do the same when removing files.

Reply:
Did you also check World Builder and the other tools? That perhaps does a bit more in terms of creating files.